lib/bolt/shell/bash.rb

```diff
@@ -390,14 +389,25 @@ def execute(command, sudoable: false, **options)
           # See if we can read from out or err, or write to in
           ready_read, ready_write, = select(read_streams.keys, write_stream, nil, timeout)
 
-          # Read from out and err
           ready_read&.each do |stream|
+            stream_name = stream == out ? 'out' : 'err'
             # Check for sudo prompt
-            read_streams[stream] << if use_sudo
-                                      check_sudo(stream, inp, options[:stdin])
-                                    else
-                                      stream.readpartial(CHUNK_SIZE)
-                                    end
+            to_print = if use_sudo
+                         check_sudo(stream, inp, options[:stdin])
+                       else
+                         stream.readpartial(CHUNK_SIZE)
+                       end
+
+            if !to_print.chomp.empty? && @stream_logger
+              formatted = ""
+              to_print.split("\\n").each_with_index do |msg, i|
+                indent = i == 0 ? '' : '  '
+                formatted += "[#{@target.safe_name}] #{stream_name}: #{indent}#{msg.chomp}\n"
+              end
+              @stream_logger.warn(formatted)
```

Comment:
```suggestion
              to_print.lines.each_with_index do |msg, i|
                indent = i == 0 ? '' : '  '
                formatted += "[#{@target.safe_name}] #{stream_name}: #{indent}#{msg}"
              end
              @stream_logger.warn(formatted.chomp)
```

I think this will fix avoid any extra newlines. Before change:

```shell
$ be bolt command run 'for v in 1 2 3 4 5; do sleep 1; echo "$v"; done' -t nix --stream
Started on nix...
The 'stream' option is experimental, and might include breaking changes between minor versions. [ID: stream_experimental]
[nix] out: 1

[nix] out: 2

[nix] out: 3

[nix] out: 4

[nix] out: 5

Finished on nix:
  STDOUT:
    1
    2
    3
    4
    5
```

With change:

```shell
$ be bolt command run 'for v in 1 2 3 4 5; do sleep 1; echo "$v"; done' -t nix --stream
Started on nix...
The 'stream' option is experimental, and might include breaking changes between minor versions. [ID: stream_experimental]
[nix] out: 1
[nix] out: 2
[nix] out: 3
[nix] out: 4
[nix] out: 5
Finished on nix:
  STDOUT:
    1
    2
    3
    4
    5
```

Reply:
Ah nice - I always forget about `lines`